### PR TITLE
feat: Deep connector Zendesk Support (Read)

### LINKED ISSUE #347: Intercom connector: Read

### PR DESCRIPTION
**Pull request by Cobalt0s**
Closes https://github.com/amp-labs/connectors/issues/616
- [x] Linter ok
- [x] Tests passing
- [x] Manual testing

## Manual `read`

Listing users.
image

### CODE REVIEW

**Comment by laurenzlong** [comment]
Hey @Cobalt0s this PR is failing to build due to " zendesksupport/connector.go:38:59: cannot use &params.Workspace (value of type *paramsbuilder.Workspace) as *map[string]string value in argument to providers.ReadInfo"

**Review by jlimatampersand — COMMENTED** [review]
Looks good overall. Left a few comments. 

Let's wait until the upstream PR gets merged and see green on build/linter etc.

File: zendesksupport/connector.go
Comment (Cobalt0s) [reply zendesksupport/connector.go:13]: Going over OpenAPI there are almost no other API versions. There are only 3 `/api/lotus/` endpoints. But they are not relevant for READ operation of a connector.

```
/api/lotus/assignables/autocomplete.json
/api/lotus/assignables/groups.json
/api/lotus/assignables/groups/{group_id}/agents.json
```

Let me know if the catalog config should have it then?
image

File: zendesksupport/parse.go
Comment (Cobalt0s) [reply zendesksupport/parse.go]: It turns out I have been doing this mistake over and over again. There 18 occurences that I will replace.

**Comment by Cobalt0s** [comment]
* **#631** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/631?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#630** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/630?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Cobalt0s and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: zendesksupport/connector.go
Comment (laurenzlong) [reply zendesksupport/connector.go:13]: I agree with Jay, in `providers/zendesk.go`, BaseURL should be modified to be `https://{{.workspace}}.zendesk.com/api`, and then this apiVersion variable should only be `v2`

**Review by laurenzlong — APPROVED** [review]

File: zendesksupport/connector.go
Comment (Cobalt0s) [reply zendesksupport/connector.go:13]: Addressed the issue. Verified Read method.
image

**Pull request merged**
